pkg/engine/armresources.go

```diff
@@ -34,6 +34,13 @@ func GenerateARMResources(cs *api.ContainerService) []interface{} {
 		armResources = append(armResources, userAssignedID, msiRoleAssignment)
 	}
 
+	if !cs.Properties.OrchestratorProfile.IsPrivateCluster() &&
+		cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku {
+		publicIPAddress := CreatePublicIPAddress(false)
+		loadBalancer := CreateAgentLoadBalancer(cs.Properties, true)
```

Comment:
what does this do? does it add LB and public IP into the templates? this will break AKS since we already generated it. I think we pass a LB resource Id, in that case we can skip the creation.

Reply:
yes we create the SLB in AKS so we expect "patch" only behavior from others(aks-e or k8s  cloud provider)

Reply:
yes `CreatePublicIPAddress` and `CreateAgentLoadBalancer` will create the PIP and the LB for the agent nodes. 

> I think we pass a LB resource Id

I could update this so that if the user provides the resource IDs for these in the apimodel, then skip creation. NOTE the `agentLbName` has to be the same as the cluster-name as the k8s cloud provider will add new loadbalancer rules to this loadbalancer for each k8s service. 

WDYT?

Reply:
the plan looks good to me.  Regarding the agentLBName, AKS already take care that by using the cluster name ("kubernetes" by default) so it should be fine.

Reply:
OK, sounds good to me. Just make sure it's clear - when client specifies `LoadBalancerBackendAddressPoolIDs` in `AgentPoolProfile", then we will skip creating the LB and any public IP. If not the case, then we need a different flag to indicate client manages these resource, so skip.

Reply:
For multiple node pools, do you guys pass in `LoadBalancerBackendAddressPoolIDs` for each pool profile and do you validate they are the same?

Reply:
Yep, they must be the same.